src/overlays/actors/ovl_En_Insect/z_en_insect.c

```diff
@@ -255,36 +274,36 @@ void func_80A7C3F4(EnInsect* this, GlobalContext* globalCtx) {
 
     SkelAnime_Update(&this->skelAnime);
     this->actor.shape.rot.y = this->actor.world.rot.y;
-    if (this->unk_31A <= 0) {
-        func_80A7C598(this);
+    if (this->actionTimer <= 0) {
+        EnInsect_SetupCrawlToSpawn(this);
     }
 
-    if (((this->unk_314 & 4) && this->unk_31C <= 0) ||
-        ((sp2E == 2 || sp2E == 3) && (this->unk_314 & 1) && (this->actor.bgCheckFlags & 1) && D_80A7DEB8 >= 4)) {
-        func_80A7CBC8(this);
-    } else if ((this->unk_314 & 1) && (this->actor.bgCheckFlags & 0x40)) {
-        func_80A7CE60(this);
+    if (((this->flags & INSECT_FLAG_TEMP) && this->lifeTimer <= 0) ||
+        (IS_DROPPED(type) && (this->flags & INSECT_FLAG_TEMP_AND_ALIVE) && (this->actor.bgCheckFlags & 1) && sDroppedCount >= 4)) {
+        EnInsect_SetupDig(this);
+    } else if ((this->flags & INSECT_FLAG_TEMP_AND_ALIVE) && (this->actor.bgCheckFlags & 0x40)) {
+        EnInsect_SetupSink(this);
     } else if (this->actor.xzDistToPlayer < 40.0f) {
-        func_80A7C818(this);
+        EnInsect_SetupRunFromPlayer(this);
     }
 }
 
-void func_80A7C598(EnInsect* this) {
-    this->unk_31A = Rand_S16Offset(10, 45);
-    func_80A7BF58(this);
-    this->actionFunc = func_80A7C5EC;
-    this->unk_314 |= 0x100;
+void EnInsect_SetupCrawlToSpawn(EnInsect* this) {
+    this->actionTimer = Rand_S16Offset(10, 45);
+    EnInsect_SetupAnimation(this);
+    this->actionFunc = EnInsect_CrawlToSpawn;
+    this->flags |= INSECT_FLAG_CRAWLING;
 }
 
-void func_80A7C5EC(EnInsect* this, GlobalContext* globalCtx) {
+void EnInsect_CrawlToSpawn(EnInsect* this, GlobalContext* globalCtx) {
```

Comment:
Just reading the code, I don't think the intent here is crawling to home (spawn), it's probably just to avoid the bugs running away too far

I mean, with this action function the bugs will hover around the home point, not reach home and stop there

In `EnInsect_Crawl` the line `Math_SmoothStepToF(&this->actor.speedXZ, 0.0f, 0.1f, 0.5f, 0.0f);` lowers the speed to 0
So I think the actual movement is handled by `EnInsect_CrawlToSpawn`, and `EnInsect_SetupCrawl` is more like "slow down"/"stop"